addOns/ascanrules/src/main/java/org/zaproxy/zap/extension/ascanrules/utils/HtmlContextAnalyser.java

```diff
@@ -0,0 +1,224 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Copyright 2011 The ZAP Development Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.zap.extension.ascanrules.utils;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import net.htmlparser.jericho.Attribute;
+import net.htmlparser.jericho.Element;
+import net.htmlparser.jericho.Source;
+import org.parosproxy.paros.network.HttpMessage;
+
+public class HtmlContextAnalyser {
+
+    private char[] quotes = {'\'', '"'};
+
+    // Tag attributes which can contain javascript
+    private String[] scriptAttributes = {
+            "onBlur",
```

Comment:
While moving from core some indentation changes were done, better to amend the commit to keep the previous indentation (similar comment for the other class, has a the moved import).